### PR TITLE
feat: implement geistoid

### PR DESCRIPTION
**Pull request by dmikalova**
> [!NOTE]
> **Medium Risk**
> Adds a batch of new card scripts that exercise complex engine paths (conditional actions, lasting control changes, discard ordering with Scrap, and global play/use prevention), which could expose edge cases in game action sequencing. Changes are mostly additive but touch high-interaction mechanics like `untilPlayerNextTurnStart` and multi-step `sequential` actions.
> 
> **Overview**
> Implements a new set of Geistoid-themed cards under `server/game/cards/14-DM`, covering a mix of *play/action/reap/fight/scrap/omni* abilities (e.g., haunted/overwhelmed conditionals, purge/capture/damage, temporary control change, and shuffling cards between zones).
> 
> Adds comprehensive specs for each new card under `test/server/cards/14-DM`, including edge cases like empty zones, opponent absence, destruction replacement, and interactions with ordered discard/Scrap resolution (with one skipped test documenting a known engine limitation).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit db2725b67c0083f308475df82f542861443c4de6. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: server/game/cards/14-DM/Dammater.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                    mode: 'exactly',
+                    numCards: 2,
+                    gameAction: ability.actions.discard()
+                },
+                then: {
+                    alwaysTriggers: true,
+                    target: {
+                        location: 'hand',
+                        controller: 'self',
+                        gameAction: ability.actions.archive()
```
Comment (strib) [inline server/game/cards/14-DM/Dammater.js:24]: This one will be missing a log message.

File: server/game/cards/14-DM/GhostTown.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    // Action: If you are haunted, archive a card. Otherwise, discard a card.
+    setupCardAbilities(ability) {
+        this.action({
+            gameAction: ability.actions.conditional({
+                condition: (context) => context.player.isHaunted(),
+                trueGameAction: ability.actions.archive((context) => ({
+                    promptForSelect: {
+                        location: 'hand',
+                        controller: 'self',
+                        message: '{0} uses {1} to archive {2}',
```
Comment (strib) [inline server/game/cards/14-DM/GhostTown.js]: This prints the archived card's name.

File: server/game/cards/14-DM/GhostTown.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                        location: 'hand',
+                        controller: 'self',
+                        message: '{0} uses {1} to discard {2}',
+                        messageArgs: (cards) => [context.player, context.source.name, cards]
+                    }
+                }))
+            }),
+            effect: '{1}',
+            effectArgs: (context) => [
+                context.player.isHaunted() ? 'archive a card' : 'discard a card'
```
Comment (strib) [inline server/game/cards/14-DM/GhostTown.js]: Do we need this message too?  Seems redundant.

File: server/game/cards/14-DM/TechnoShade.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    setupCardAbilities(ability) {
+        this.reap({
+            fight: true,
+            condition: (context) =>
+                !!context.player.opponent && context.player.opponent.hand.length > 0,
+            gameAction: ability.actions.returnToDeck((context) => {
+                const hand = context.player.opponent.hand;
+                return {
+                    shuffle: true,
+                    target: hand[Math.floor(Math.random() * hand.length)]
```
Comment (strib) [inline server/game/cards/14-DM/TechnoShade.js]: We use `_.shuffle().slice(0,1)` in other similar places in the code (like Eddy of Dis).

File: server/game/cards/14-DM/VarghastsVengeance.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                        )
+                    ]
+                })),
+                ability.actions.gainChains({ amount: 2 })
+            ]
+        });
+
+        this.scrap({
+            condition: (context) => !!context.player.opponent,
+            effect: "purge a card from {1}'s discard pile and have them shuffle their discard pile into their deck",
```
Comment (strib) [inline server/game/cards/14-DM/VarghastsVengeance.js]: We should print the card name.

**Review by strib — APPROVED** [review]

File: server/game/cards/14-DM/GhostTown.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+                        controller: 'self',
+                        message: '{0} uses {1} to archive a card',
+                        messageArgs: [context.player, context.source]
+                    }
+                }),
+                falseGameAction: ability.actions.discard({
+                    promptForSelect: {
+                        location: 'hand',
+                        controller: 'self'
+                    }
```
[the remark this answers is not in the document]
Comment (strib) [reply server/game/cards/14-DM/GhostTown.js:22]: This one is over-compensating, since the engine already prints a discard message.

File: server/game/cards/14-DM/TechnoShade.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+            fight: true,
+            condition: (context) =>
+                !!context.player.opponent && context.player.opponent.hand.length > 0,
+            gameAction: ability.actions.returnToDeck((context) => ({
+                shuffle: true,
+                target: _.shuffle(
+                    context.player.opponent ? context.player.opponent.hand : []
+                ).slice(0, 1)
+            }))
+        });
```
[the remark this answers is not in the document]
Comment (strib) [reply server/game/cards/14-DM/TechnoShade.js:18]: Obviously wrong.
